### PR TITLE
[AIRFLOW-5156] Added auth type to HttpHook

### PR DESCRIPTION
**Pull request by randr97**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Nice one!

File: airflow/providers/http/hooks/http.py [hunk 1]
```diff
@@ -27,20 +28,28 @@ class HttpHook(BaseHook):
...
+    :param method: the API method to be called
+    :type method: str
     :param http_conn_id: connection that has the base API url i.e https://www.google.com/
         and optional authentication credentials. Default headers can also be specified in
         the Extra field in json format.
     :type http_conn_id: str
-    :param method: the API method to be called
-    :type method: str
+    :param auth_type: the API method to be called
+    :type method: AuthBase of python requests lib
```
Comment (eladkal) [inline airflow/providers/http/hooks/http.py:38]: It should be 
`    :type auth_type: AuthBase of python requests lib`

File: airflow/providers/http/hooks/http.py [hunk 2]
```diff
@@ -66,7 +75,7 @@ def get_conn(self, headers=None):
             if conn.port:
                 self.base_url = self.base_url + ":" + str(conn.port)
             if conn.login:
-                session.auth = (conn.login, conn.password)
+                session.auth = self.auth_type(conn.login, conn.password)
```
Comment (mik-laj) [inline airflow/providers/http/hooks/http.py:78]: Other auth methods require a different number of arguments, e.g. OAuth2.What is the purpose of this change?

Comment (jeffolsi) [reply airflow/providers/http/hooks/http.py:78]: the change make sense to me since it uses request package so it can work with custom auth
https://2.python-requests.org/en/master/user/advanced/#custom-authentication
provide the 
the auth_type in this context is auth object passed in the operator constructor

Comment (mik-laj) [reply airflow/providers/http/hooks/http.py:78]: https://requests.readthedocs.io/en/master/user/authentication/
Is it possible to use this hook with other methods? I think it would be better if the user passed the auth object, not just the auth_type.

Comment (randr97) [reply airflow/providers/http/hooks/http.py:78]: @mik-laj the reason why I have given a default value is cus of backward compatibility.
For other auth methods I think we will have to build stuff out.
